src/engine/positionscratchcontroller.cpp

```diff
@@ -199,27 +199,18 @@ void PositionScratchController::process(double currentSamplePos,
             // sure.
             m_inertiaEnabled = false;
 
-            double sampleDelta = 0.0;
+            double sampleDelta = currentSamplePos - m_prevSamplePos;
             if (wrappedAround > 0) {
-                // If we wrapped around calculate the virtual position like if
-                // we are not looping, i.e. sum up diffs from loop start/end and
-                // loop length for each wrap-aound (necessary if the buffer is
-                // longer than the loop, e.g. when looping at high rates / with short loops.
-                // This avoids high rate and infinite wrap-around scratching
-                // even after mouse was stopped.
+                // If we wrapped around, add  loop length for each wrap-aound.
+                // This is necessary if the buffer is longer than the loop,
+                //  e.g. when looping at high rates / with short loops.
                 double triggerPos = trigger.toEngineSamplePos();
                 double targetPos = target.toEngineSamplePos();
-                bool reverse = triggerPos < targetPos;
-                double loopLength = reverse ? -1 * (targetPos - triggerPos)
-                                            : triggerPos - targetPos;
-                if (wrappedAround > 2) {
-                    sampleDelta = (wrappedAround - 1) * loopLength;
+                double loopLength = triggerPos - targetPos;
+                if (loopLength < 0) {
+                    loopLength *= -1;
```

Comment:
This is not necessary, the trigger / target returned by [`LoopingControl::nextTrigger`](https://github.com/mixxxdj/mixxx/blob/4c1b8813fba2029047dc06656af25d6e58ea6d52/src/engine/controls/loopingcontrol.cpp#L410) are already swapped in case of reverse.